### DISCUSSION TITLE
How can I contribute from a Tasmota-IR device?

### DISCUSSION THREAD

**Discussion by voltagex** [probonopd/irdb#67] [Q&A]
Hi,
This seems like it could be quite a useful project. I have some remotes I'd like to decode and eventually use in my home automation. I have several commercial IR transceivers that have been flashed with Tasmota-IR

The output when they receive a signal (as of version 9) looks like `{"IrReceived":{"Protocol":"NEC","Bits":32,"Data":"0x482C609F","DataLSB":"0x123406F9","Repeat":0,"RawData":"+8955-4500+555-565+550-1695E-570EdEfEdEdE-580EdEdEfEdEfEfEdEhEdEfEfEgEgEdEdEhEfEdEgEfEfEfEfE-1700E-43040+8950-2255E","RawDataInfo":[71,71,0]}}` and `{"IrReceived":{"Protocol":"NEC","Bits":0,"Data":"0xFFFFFFFFFFFFFFFF","DataLSB":"0xFFFFFFFFFFFFFFFF","Repeat":1,"RawData":"+9060-2285+560","RawDataInfo":[3,3,0]}}`

which is (almost) documented https://tasmota.github.io/docs/IRSend-RAW-Encoding/#new-ir-raw-compact-encoding

While I will eventually build an Arduino-based receiver, I'd really like to use what I have. How do I go from this compact format to something useful for your project?

**Comment by probonopd** [comment]
This seems to be what Tasmota calls "IRSend IR Raw compact encoding" (available since v9.0.0.1): https://tasmota.github.io/docs/IRSend-RAW-Encoding/ 

If you need to transform the compact format in the legacy format, you can use the online tool: https://tasmota.hadinger.fr/util

The tool says that our code above is eqivalent to

```
8955,4500,555,565,550,1695,550,570,550,565,550,1695,550,565,550,565,550,580,550,565,550,565,550,1695,550,565,550,1695,550,1695,550,565,550,580,550,565,550,1695,550,1695,550,570,550,570,550,565,550,565,550,580,550,1695,550,565,550,570,550,1695,550,1695,550,1695,550,1695,550,1700,550,43040,8950,2255,550
```

From here you should be able to use IrScrutiinizer to decode the signal into something that can be put into irdb.

Maybe @bengtmartensson will conider to add Tasmota support natively one day?
https://github.com/bengtmartensson/IrScrutinizer/issues/356

**Comment by bengtmartensson** [comment]
@voltagex : If I understand you correctly, you want to use your Tasmota device to "record" remotes in your possession, to create collections of named commands, to be saved, for example in the CVS format of the current project. It is not the question of importing existing such collections in "Tasmota format".

I do not know the answer to that question; you should direct it to the Tasmota community. 

However, individual signals is simple: The "old" format is nothing but the raw durations in micro seconds, with commas in between. The "new" format is not very hard to decode (it is actually very close to GlobalCache's compressed format), and can be converted as @probonopd  described. With IrScrutinizer, these irsignals can be imported and manipulated, and for example exported to CVS format.

Would [this](http://www.harctoolbox.org/arduino_nano.html) be an alternative, using AGirs/Infrared4Arduino?
